src/main/java/pl/futurecollars/invoicing/db/memory/InMemoryDatabase.java

```diff
@@ -5,9 +5,13 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
+import org.springframework.context.annotation.Primary;
+import org.springframework.stereotype.Repository;
 import pl.futurecollars.invoicing.db.Database;
 import pl.futurecollars.invoicing.model.Invoice;
 
+@Primary
```

Comment:
Skoro korzystasz z @Qualifier do wstrzyknięcia odpowiedniej bazy danych to @Primary nie ma zastosowania.